src/components/user-list.js

```diff
@@ -11,52 +11,60 @@ export default class UserList extends React.Component {
         super(props);
 
         this.state = this._getAppState();
-
-        this.handleClickAddUser = this.handleClickAddUser.bind(this);
     }
 
     _getAppState () {
         return {
-            wantAdd:  false,
-            wantEdit: false,
-            users:    UserStore.getUsers()
+            duplicateUserId: '',
+            addDuplicate:    false,
+            wantAdd:         false,
+            users:           UserStore.getUsers()
         };
     }
 
-    handleClickAddUser () {
-        this.setState({ wantAdd: !this.state.wantAdd });
-    }
-
     _onChange = () => {
         this.setState(this._getAppState());
     };
 
+    _handleClickAddUser = () => {
+        this.setState({ wantAdd: !this.state.wantAdd });
+    };
+
+    _addingFailed = userId => {
+        let { addDuplicate } = this.state;
+
+        this.setState({ duplicateUserId: userId, addDuplicate: !addDuplicate });
+    };
+
     componentDidMount () {
-        UserStore.on(CHANGE_EVENT, this._onChange);
+        UserStore.on(EVENT_TYPE.change, this._onChange);
+        UserStore.on(EVENT_TYPE.userAdded, this._onChange);
+        UserStore.on(EVENT_TYPE.addingFailed, this._addingFailed);
     }
 
     componentWillUnmount () {
-        UserStore.removeListener(CHANGE_EVENT, this._onChange);
+        UserStore.off(EVENT_TYPE.change, this._onChange);
+        UserStore.off(EVENT_TYPE.userAdded, this._onChange);
+        UserStore.off(EVENT_TYPE.addingFailed, this._addingFailed);
     }
 
     render () {
-        const { wantAdd, users } = this.state;
+        const { wantAdd, users, duplicateUserId, addDuplicate } = this.state;
 
         return (
             <div className="UserList">
-                <button className="ButtonAddUser" data-testid="ButtonAddUser" onClick={this.handleClickAddUser}>
+                <button className="ButtonAddUser" data-testid="ButtonAddUser" onClick={this._handleClickAddUser}>
                     Add new User
                 </button>
                 {
                     wantAdd &&
                     (
-                        <User isNewUser={true}/>
+                        <User isNewUser={true} duplicateUserId={duplicateUserId} addDuplicate={addDuplicate}/>
```

Comment:
Seems like `user` don't use `addDuplicate` from props and has its own one flag `addDuplicate`.
Do we need it?